pmm-app/tests/pages/qanPage.js

```diff
@@ -137,13 +137,18 @@ module.exports = {
 
   getFilterLocator(filterValue) {
     const filterLocator =
-     "//section[@class='aside__filter-group']//span[contains(text(), '" +
-     filterValue +
-     "')]/../span[@class='checkbox-container__checkmark']";
+      "//section[@class='aside__filter-group']//span[contains(text(), '" +
```

Comment:
The general impression I get when reading this PR is that our locators are not sustainable and not resistant to change, so any minor change  in html (with no change in the component logic) would require to update the locators. We should just leverage `data-qa` or `data-test-id` properties. I would encourage to upgrade our components to provide a way to universally expose those props to make it possible.

The other best practice is to prefer css-selectors over xml-selectors as being more resilient.

Reply:
Good example: https://github.com/percona/grafana-dashboards/pull/715
No logic changed there, but the tests failed.